src/Orleans/Telemetry/Consumers/FileTelemetryConsumer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace Orleans.Runtime
+{
+    public class FileTelemetryConsumer : ITraceTelemetryConsumer
+    {
+        private StreamWriter _logOutput;
+
+        public FileTelemetryConsumer(string fileName) : this(new FileInfo(fileName))
+        {
+        }
+
+        public FileTelemetryConsumer(FileInfo file)
+        {
+            var fileExists = File.Exists(file.FullName);
+            _logOutput = fileExists ? file.AppendText() : file.CreateText();
+            file.Refresh();
+        }
+
+        public void TrackTrace(string message)
+        {
+            if (_logOutput == null) return;
+            _logOutput.WriteLine(message);
```

Comment:
We definitely need a lock here. It races and throws in functional tests.
>Exc level 0: System.IndexOutOfRangeException: Probable I/O race condition detected while copying memory. The I/O package is not thread safe by default. In multithreaded applications, a stream must be accessed in a thread-safe way, such as a thread-safe wrapper returned by TextReader's or TextWriter's Synchronized methods. This also applies to classes like StreamWriter and StreamReader.
   at System.Buffer.InternalBlockCopy(Array src, Int32 srcOffsetBytes, Array dst, Int32 dstOffsetBytes, Int32 byteCount)
   at System.IO.StreamWriter.Write(Char[] buffer, Int32 index, Int32 count)
   at Orleans.Runtime.FileTelemetryConsumer.TrackTrace(String message) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\Orleans\Telemetry\Consumers\FileTelemetryConsumer.cs:line 25
   at Orleans.Runtime.TraceLogger.TrackTrace(String message, Severity severity) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\Orleans\Logging\TraceLogger.cs:line 1147
   at Orleans.Runtime.TraceLogger.WriteLogMessage(Int32 errorCode, Severity sev, String format, Object[] args, Exception exception) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\Orleans\Logging\TraceLogger.cs:line 814
   at Orleans.Runtime.TraceLogger.Info(ErrorCode errorCode, String format, Object[] args) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\Orleans\Logging\TraceLogger.cs:line 636
   at Orleans.Runtime.SystemStatus.set_Current(SystemStatus value) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\Orleans\Runtime\SystemStatus.cs:line 51
   at Orleans.Runtime.Silo.Terminate(Boolean gracefully) in c:\Builds\5877\Orleans\sbykov-Orleans-VSO-BVT-GH-Func\src\src\OrleansRuntime\Silo\Silo.cs:line 660
   at Orleans.Runtime.Silo.Shutdown()